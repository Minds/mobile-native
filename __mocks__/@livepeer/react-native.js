// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
