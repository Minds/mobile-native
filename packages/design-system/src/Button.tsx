import React, { forwardRef } from 'react';
import { TamaguiElement, styled, themeable, GetProps } from '@tamagui/core';
import { getButtonSized } from '@tamagui/get-button-sized';
import { ThemeableStack } from '@tamagui/stacks';
import { SizableText } from '@tamagui/text';
import { ButtonProps, useButton } from './buttonHelpers';

const NAME = 'Button';

export const ButtonFrame = styled(ThemeableStack, {
  name: NAME,
  tag: 'button',
  focusable: true,
  hoverTheme: true,
  pressTheme: true,
  backgrounded: true,
  borderWidth: 1,
  borderColor: 'transparent',
  justifyContent: 'center',
  alignItems: 'center',
  flexWrap: 'nowrap',
  flexDirection: 'row',
  borderRadius: 10000,

  // if we wanted this only when pressable = true, we'd need to merge variants?
  cursor: 'pointer',

  pressStyle: {
    borderColor: 'transparent',
  },

  hoverStyle: {
    borderColor: 'transparent',
  },

  focusStyle: {
    borderColor: '$borderColorFocus',
  },

  variants: {
    size: {
      '...size': getButtonSized,
    },
    sSize: {
      xl: (_, extra) => getButtonSized('$5', extra),
      l: (_, extra) => getButtonSized('$4', extra),
      m: (_, extra) => getButtonSized('$3.5', extra),
      s: (_, extra) => getButtonSized('$3', extra),
    },
    disabled: {
      true: {
        pointerEvents: 'none',
        opacity: 0.5,
      },
    },
  } as const,

  defaultVariants: {
    size: '$3',
  },
});

export const ButtonText = styled(SizableText, {
  name: NAME,
  userSelect: 'none',
  cursor: 'pointer',
  // flexGrow 1 leads to inconsistent native style where text pushes to start of view
  flexGrow: 0,
  flexShrink: 1,
  ellipse: true,

  variants: {
    sSize: {
      xl: {
        size: '$5',
      },
      l: {
        size: '$4',
      },
      m: {
        size: '$3.5',
      },
      s: {
        size: '$3',
      },
    },
  } as const,
  defaultVariants: {
    size: '$3.5',
  },
});

type ButtonFrameProps = GetProps<typeof ButtonFrame>;
const ButtonComponent = forwardRef<TamaguiElement, ButtonFrameProps>(
  (props, ref) => {
    const { props: buttonProps } = useButton(props as ButtonProps, ButtonText);
    return <ButtonFrame {...buttonProps} ref={ref} />;
  },
);

export const buttonStaticConfig = {
  inlineProps: new Set([
    // text props go here (can't really optimize them, but we never fully extract button anyway)
    // may be able to remove this entirely, as the compiler / runtime have gotten better
    'color',
    'fontWeight',
    'fontSize',
    'fontFamily',
    'letterSpacing',
    'textAlign',
  ]),
};

export const Button = ButtonFrame.extractable(
  themeable(ButtonComponent, ButtonFrame.staticConfig),
  buttonStaticConfig,
);
