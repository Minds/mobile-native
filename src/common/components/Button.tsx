import React, { Component } from 'react';
import {
  Text,
  TouchableOpacity,
  TextStyle,
  ViewStyle,
  GestureResponderEvent,
  TouchableOpacityProps,
  Platform,
} from 'react-native';
import { DotIndicator } from 'react-native-reanimated-indicators';

import ThemedStyles from '../../styles/ThemedStyles';

export interface ButtonPropsType extends TouchableOpacityProps {
  text?: string;
  loading?: boolean;
  onPress?: (ev: GestureResponderEvent) => void;
  textColor?: string;
  color?: string;
  children?: React.ReactNode;
  containerStyle?: ViewStyle | Array<ViewStyle>;
  accessibilityLabel?: string;
  textStyle?: TextStyle | Array<TextStyle>;
  disabled?: boolean;
  inverted?: boolean;
  large?: boolean;
  small?: boolean;
  xSmall?: boolean;
  transparent?: boolean;
  action?: boolean;
  active?: boolean;
  borderless?: boolean;
}

/**
 * Custom Button component
 */
export default class Button extends Component<ButtonPropsType> {
  /**
   * Default props
   */
  static defaultProps = {
    loading: false,
  };

  /**
   * Render
   */
  render() {
    const theme = ThemedStyles.style;

    const {
      onPress,
      textColor,
      color,
      children,
      containerStyle,
      accessibilityLabel,
      textStyle,
      disabled,
      loading,
      inverted,
      large,
      small,
      xSmall,
      transparent,
      action,
      active,
      borderless,
      ...extraProps
    } = this.props;

    let background = ThemedStyles.getColor(
      active ? 'active' : 'primary_background',
    );
    let mainColor =
      color || transparent ? '#FFFFFF' : ThemedStyles.getColor('primary_text');

    if (inverted !== undefined) {
      background = mainColor;
      mainColor = ThemedStyles.getColor('primary_background');
    }

    const padding = {
      paddingVertical: small || xSmall ? 8 : 10,
      paddingHorizontal: large ? 23 : xSmall ? 16 : 21,
    };

    const border = borderless
      ? {}
      : {
          ...(action ? theme.borderLink : theme.borderPrimary),
          ...theme.border,
        };

    const transparentStyle = transparent
      ? {
          backgroundColor: 'rgba(0,0,0,0.40)',
          borderColor: action
            ? ThemedStyles.getColor('link')
            : Platform.select({
                android: 'rgba(255,255,255,0.40)',
                ios: 'rgba(255,255,255,0.60)',
              }),
        }
      : {};

    const style = {
      backgroundColor: background,
      borderRadius: 21,
      ...padding,
      ...border,
      ...transparentStyle,
    };

    const fontSize = { fontSize: large ? 19 : small ? 16 : xSmall ? 14 : 17 };
    const body = loading ? (
      <DotIndicator
        containerStyle={[theme.rowJustifyCenter, theme.padding2x]}
        color={mainColor}
        scaleEnabled={true}
      />
    ) : (
      this.props.text && (
        <Text style={[fontSize, { color: textColor || mainColor }, textStyle]}>
          {' '}
          {this.props.text}{' '}
        </Text>
      )
    );

    const onButtonPress = loading ? undefined : onPress;

    return (
      <TouchableOpacity
        onPress={onButtonPress}
        disabled={disabled}
        accessibilityLabel={accessibilityLabel}
        style={[theme.rowJustifyCenter, theme.centered, style, containerStyle]}
        {...extraProps}>
        {children}
        {body}
      </TouchableOpacity>
    );
  }
}
