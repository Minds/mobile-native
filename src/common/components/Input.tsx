//@ts-nocheck
import React, { Component } from 'react';
import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity,
  TextInput,
  TextStyle,
  TextInputProps,
} from 'react-native';
import DateTimePicker from 'react-native-modal-datetime-picker';
import InfoPopup from './InfoPopup';

import ThemedStyles from '../../styles/ThemedStyles';
import PhoneValidationComponent from './phoneValidation/PhoneValidationComponent';

export interface PropsType extends TextInputProps {
  TFA?: any;
  TFAConfirmed?: boolean;
  inputType?: string;
  optional?: boolean;
  autofocus?: boolean;
  dateFormat?: string;
  labelStyle?: TextStyle | Array<TextStyle>;
  placeholder?: string;
  value?: string;
  testID?: string;
  keyboardType?: string;
  editable?: boolean;
  scrollEnabled?: boolean;
  secureTextEntry?: boolean;
  multiline?: boolean;
  selectTextOnFocus?: boolean;
  onChangeText?: (string) => void;
  onFocus?: () => void;
  onBlur?: () => void;
  onEndEditing?: () => void;
  onSubmitEditing?: () => void;
  style?: any;
  info?: string;
  error?: string;
}

/**
 * Form input
 */
export default class Input extends Component<PropsType> {
  timeoutCleanup = null;
  /**
   * State
   */
  state = {
    datePickerVisible: false,
  };

  /**
   * Constructor
   * @param props
   */
  constructor(props) {
    super(props);
    this.inputRef = React.createRef();
  }

  componentDidMount() {
    this.shouldAutofocus();
  }

  shouldAutofocus() {
    this.timeoutCleanup = setTimeout(() => {
      if (this.inputRef.current && this.props.autofocus) {
        this.inputRef.current.focus();
      }
    }, 300);
  }

  componentWillUnmount() {
    if (this.timeoutCleanup) {
      clearTimeout(this.timeoutCleanup);
    }
  }

  /**
   * Show date picker
   */
  showDatePicker = () => {
    this.setState({ datePickerVisible: true });
  };

  /**
   * Dismiss date picker
   */
  dismissDatePicker = () => {
    this.setState({ datePickerVisible: false });
  };

  /**
   * Confirm date picker
   */
  confirmDatePicker = (date) => {
    let dateString = '';
    switch (this.props.dateFormat) {
      case 'ISOString':
        dateString = date.toISOString().substring(0, 10);
        break;
      default:
        dateString = date.toLocaleDateString();
        break;
    }
    this.dismissDatePicker();
    this.props.onChangeText(dateString);
  };

  /**
   * Text input
   */
  textInput = () => {
    const theme = ThemedStyles.style;
    return (
      <TextInput
        {...this.props}
        style={[theme.input, this.props.style]}
        placeholderTextColor="#444"
        returnKeyType={'done'}
        autoCapitalize={'none'}
        underlineColorAndroid="transparent"
        ref={this.inputRef}
        placeholder=""
      />
    );
  };

  /**
   * Phone input
   */
  phoneInput = () => {
    const theme = ThemedStyles.style;
    return (
      <PhoneValidationComponent
        style={[theme.input, this.props.style]}
        textStyle={theme.colorPrimaryText}
        onFocus={this.props.onFocus}
        onBlur={this.props.onBlur}
        TFA={this.props.TFA}
        TFAConfirmed={this.props.TFAConfirmed}
      />
    );
  };

  /**
   * Date input
   */
  dateInput = () => {
    const theme = ThemedStyles.style;
    const maxDate = new Date();
    maxDate.setFullYear(maxDate.getFullYear() - 13);
    return (
      <View>
        <TouchableOpacity
          {...this.props}
          style={[theme.input, this.props.style]}
          placeholderTextColor="#444"
          returnKeyType={'done'}
          autoCapitalize={'none'}
          underlineColorAndroid="transparent"
          placeholder=""
          onPress={this.showDatePicker}>
          <Text style={theme.colorPrimaryText}>{this.props.value}</Text>
        </TouchableOpacity>
        <DateTimePicker
          isVisible={this.state.datePickerVisible}
          onConfirm={this.confirmDatePicker}
          date={maxDate}
          maximumDate={maxDate}
          onCancel={this.dismissDatePicker}
          mode="date"
          display="spinner"
        />
      </View>
    );
  };

  /**
   * renders correct input
   */
  renderInput = () => {
    const inputType = this.props.inputType;
    if (inputType) {
      switch (inputType) {
        case 'textInput':
          return this.textInput();
        case 'phoneInput':
          return this.phoneInput();
        case 'dateInput':
          return this.dateInput();
      }
    }
    return this.textInput();
  };

  /**
   * Render
   */
  render() {
    const theme = ThemedStyles.style;
    const optional = this.props.optional ? (
      <Text style={[styles.optional, theme.colorSecondaryText]}>
        {'Optional'}
      </Text>
    ) : null;

    return (
      <View>
        <View style={styles.row}>
          <View style={styles.row}>
            <Text
              style={[
                styles.label,
                theme.colorSecondaryText,
                this.props.labelStyle,
              ]}>
              {this.props.placeholder}
            </Text>
            {this.props.info && <InfoPopup info={this.props.info} />}
            {!!this.props.error && (
              <View style={styles.errorContainer}>
                <Text style={[theme.colorAlert, theme.fontL, theme.textRight]}>
                  {this.props.error}
                </Text>
              </View>
            )}
          </View>
          {optional}
        </View>
        {this.renderInput()}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: '100%',
  },
  label: {
    fontSize: 16,
    fontWeight: '600',
    marginBottom: 5,
    fontFamily: 'Roboto',
  },
  errorContainer: {
    alignContent: 'flex-end',
    flexGrow: 1,
    paddingRight: 10,
  },
  optional: {
    fontSize: 14,
    fontFamily: 'Roboto-Italic',
  },
  shadow: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.1,
    shadowRadius: 1,
    elevation: 1,
  },
});
