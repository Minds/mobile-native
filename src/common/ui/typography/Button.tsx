import React from 'react';
import { Typography, TypographyPropsType } from './Typography';

type BodyType = Omit<TypographyPropsType, 'type'>;

export const Btn1 = (props: BodyType) => <Typography type="Btn1" {...props} />;
export const Btn2 = (props: BodyType) => <Typography type="Btn2" {...props} />;
export const Btn3 = (props: BodyType) => <Typography type="Btn3" {...props} />;
