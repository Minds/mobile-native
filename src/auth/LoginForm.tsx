import React, { Component } from 'react';
import * as Animatable from 'react-native-animatable';

import {
  View,
  Text,
  LayoutAnimation,
  Dimensions,
  Platform,
} from 'react-native';

import Icon from 'react-native-vector-icons/Ionicons';

import authService from './AuthService';

import i18n from '../common/services/i18n.service';
import logService from '../common/services/log.service';
import Button from '../common/components/Button';
import ThemedStyles from '../styles/ThemedStyles';
import InputContainer from '../common/components/InputContainer';
import BoxShadow from '../common/components/BoxShadow';
import { styles, shadowOpt, icon } from './styles';

type PropsType = {
  onLogin?: Function;
  onRegisterPress?: () => void;
  onForgot: Function;
};

type StateType = {
  username: string;
  password: string;
  msg: string;
  twoFactorToken: string;
  twoFactorCode: string;
  hidePassword: boolean;
  inProgress: boolean;
};

const { height } = Dimensions.get('window');

const loginMargin = { marginTop: height / 55 };

/**
 * Login Form
 */
export default class LoginForm extends Component<PropsType, StateType> {
  /**
   * State
   */
  state = {
    username: '',
    password: '',
    msg: '',
    twoFactorToken: '',
    twoFactorCode: '',
    language: '',
    hidePassword: true,
    inProgress: false,
  };

  /**
   * Constructor
   */
  constructor(props: PropsType) {
    super(props);
    this.state.language = i18n.getCurrentLocale();
  }

  /**
   * Render
   */
  render() {
    const theme = ThemedStyles.style;

    const msg = this.state.msg ? (
      <Animatable.Text
        animation="bounceInLeft"
        useNativeDriver
        style={[theme.subTitleText, theme.colorSecondaryText, theme.textCenter]}
        testID="loginMsg">
        {this.state.msg}
      </Animatable.Text>
    ) : null;

    const inputs = (
      <View style={styles.shadow}>
        <InputContainer
          containerStyle={styles.inputBackground}
          labelStyle={theme.colorWhite}
          style={theme.colorWhite}
          placeholder={i18n.t('auth.username')}
          onChangeText={this.setUsername}
          autoCompleteType="username"
          textContentType="username"
          value={this.state.username}
          testID="usernameInput"
          noBottomBorder
        />
        <View>
          <InputContainer
            containerStyle={styles.inputBackground}
            labelStyle={theme.colorWhite}
            style={theme.colorWhite}
            placeholder={i18n.t('auth.password')}
            secureTextEntry={this.state.hidePassword}
            autoCompleteType="password"
            textContentType="password"
            onChangeText={this.setPassword}
            value={this.state.password}
            testID="userPasswordInput"
          />
          <Icon
            name={this.state.hidePassword ? 'md-eye' : 'md-eye-off'}
            size={25}
            onPress={this.toggleHidePassword}
            style={[theme.inputIcon, icon]}
          />
        </View>
      </View>
    );

    const inputsWithShadow = Platform.select({
      ios: inputs,
      android: <BoxShadow setting={shadowOpt}>{inputs}</BoxShadow>, // Android fallback for shadows
    });

    return (
      <View style={theme.flexContainer}>
        {msg}
        {inputsWithShadow}
        <View style={[theme.margin6x, theme.flexContainer]}>
          <Button
            onPress={() => this.onLoginPress()}
            text={i18n.t('auth.login')}
            containerStyle={[loginMargin, theme.fullWidth]}
            loading={this.state.inProgress}
            disabled={this.state.inProgress}
            accessibilityLabel="loginButton"
            transparent
            large
          />
          <View style={theme.marginTop4x}>
            <Text
              style={[theme.colorWhite, theme.fontL, theme.textCenter]}
              onPress={this.onForgotPress}>
              {i18n.t('auth.forgot')}
            </Text>
          </View>
          <View style={theme.flexContainer} />
          <Button
            onPress={this.props.onRegisterPress}
            text={i18n.t('auth.createChannel')}
            containerStyle={[theme.fullWidth, theme.marginTop6x]}
            disabled={this.state.inProgress}
            testID="registerButton"
            transparent
            large
          />
        </View>
      </View>
    );
  }

  /**
   * Set two factor
   * @param {string} value
   */
  setTwoFactor = (value) => {
    const twoFactorCode = String(value).trim();
    this.setState({ twoFactorCode });
  };

  /**
   * Set two factor
   * @param {string} value
   */
  setUsername = (value) => {
    const username = String(value).trim();
    this.setState({ username });
  };

  /**
   * Set two factor
   * @param {string} value
   */
  setPassword = (value) => {
    const password = String(value).trim();
    this.setState({ password });
  };

  /**
   * Set two factor
   * @param {string} value
   */
  toggleHidePassword = () => {
    this.setState({ hidePassword: !this.state.hidePassword });
  };

  /**
   * Handle forgot password
   */
  onForgotPress = () => {
    this.props.onForgot();
  };

  /**
   * On login press
   */
  onLoginPress() {
    this.setState({ msg: '', inProgress: true });
    // is two factor auth
    if (this.state.twoFactorToken) {
      authService
        .twoFactorAuth(this.state.twoFactorToken, this.state.twoFactorCode)
        .then(() => {
          if (this.props.onLogin) {
            this.props.onLogin();
          }
        })
        .catch((err) => {
          logService.exception('[LoginForm]', err);
        });
    } else {
      authService
        .login(this.state.username, this.state.password)
        .then(() => {
          this.props.onLogin && this.props.onLogin();
        })
        .catch((errJson) => {
          LayoutAnimation.configureNext(LayoutAnimation.Presets.spring);
          if (
            errJson.error === 'invalid_grant' ||
            errJson.error === 'invalid_client'
          ) {
            this.setState({
              msg: i18n.t('auth.invalidGrant'),
              inProgress: false,
            });
            return;
          }

          //TODO implement on backend and edit
          if (errJson.error === 'two_factor') {
            this.setState({
              twoFactorToken: errJson.message,
              inProgress: false,
            });
            return;
          }

          this.setState({
            msg: errJson.message || 'Unknown error',
            inProgress: false,
          });

          logService.exception('[LoginForm]', errJson);
        });
    }
  }
}
