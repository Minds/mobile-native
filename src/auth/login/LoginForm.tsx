import React from 'react';

import { View } from 'react-native';
import ThemedStyles from '../../styles/ThemedStyles';
import { observer, useLocalStore } from 'mobx-react';
import ResetPasswordModal, {
  ResetPasswordModalHandles,
} from '../reset-password/ResetPasswordModal';
import Icon from 'react-native-vector-icons/Ionicons';
import createLoginStore from './createLoginStore';
import FastImage from 'react-native-fast-image';
import UserModel from '../../channel/UserModel';
import sessionService from '../../common/services/session.service';
import InputContainer from '../../common/components/InputContainer';
import i18n from '../../common/services/i18n.service';
import MText from '../../common/components/MText';
import { IS_IOS } from '../../config/Config';
import { BottomSheetButton } from '../../common/components/bottom-sheet';

type PropsType = {
  onLogin?: Function;
  onRegisterPress?: () => void;
  multiUser?: boolean;
  relogin?: boolean;
  sessionIndex?: number;
};
/**
 * Login Form component
 */
export default observer(function LoginForm(props: PropsType) {
  const resetRef = React.useRef<ResetPasswordModalHandles>(null);
  const localStore = useLocalStore(createLoginStore, { props, resetRef });

  const theme = ThemedStyles.style;

  const user =
    props.sessionIndex !== undefined
      ? UserModel.checkOrCreate(
          sessionService.tokensData[props.sessionIndex].user,
        )
      : sessionService.getUser();

  const usernameInput = props.relogin ? (
    <View style={styles.container}>
      <FastImage
        source={user.getAvatarSource('medium')}
        style={styles.avatar}
      />
      <View style={styles.nameContainer}>
        <MText style={styles.name}>{user.name}</MText>
        <MText style={styles.username} testID={`username${user.username}`}>
          @{user.username}
        </MText>
      </View>
    </View>
  ) : (
    <InputContainer
      placeholder={i18n.t('auth.username')}
      onChangeText={localStore.setUsername}
      autoCompleteType="username"
      textContentType="username"
      value={localStore.username}
      testID="usernameInput"
      noBottomBorder
      autoFocus={props.multiUser}
    />
  );

  return (
    <View style={theme.flexContainer}>
      {usernameInput}
      <View style={theme.marginBottom4x}>
        <InputContainer
          placeholder={i18n.t('auth.password')}
          secureTextEntry={localStore.hidePassword}
          autoCompleteType="password"
          textContentType="password"
          onChangeText={localStore.setPassword}
          value={localStore.password}
          testID="userPasswordInput"
          autoFocus={props.relogin}
        />
        <Icon
          name={localStore.hidePassword ? 'md-eye' : 'md-eye-off'}
          size={25}
          onPress={localStore.toggleHidePassword}
          style={styles.icon}
        />
      </View>
      <BottomSheetButton
        action
        loading={localStore.inProgress}
        text={i18n.t('auth.login')}
        onPress={localStore.onLoginPress}
      />
      <View style={theme.marginTop4x}>
        <MText style={styles.forgotText} onPress={localStore.onForgotPress}>
          {i18n.t('auth.forgot')}
        </MText>
      </View>
      <ResetPasswordModal ref={resetRef} />
    </View>
  );
});

const styles = ThemedStyles.create({
  name: ['bold', 'fontXL'],
  username: ['fontMedium', 'fontM'],
  forgotText: ['colorPrimaryText', 'fontL', 'textCenter'],
  container: [
    'rowJustifyCenter',
    'padding4x',
    'bcolorPrimaryBorder',
    'borderTopHair',
    'bgPrimaryBackground',
  ],
  nameContainer: ['flexContainerCenter', 'paddingLeft', 'justifyCenter'],
  avatar: [
    {
      height: 40,
      width: 40,
      borderRadius: 20,
    },
    'bgTertiaryBackground',
  ],
  icon: [
    {
      position: 'absolute',
      right: 12,
      top: IS_IOS ? 30 : 33,
    },
    'colorSecondaryText',
  ],
});
