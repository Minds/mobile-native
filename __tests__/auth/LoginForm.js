import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import LoginForm from '../../src/auth/LoginForm';
import authService from '../../src/auth/AuthService';
import Input from '../../src/common/components/Input';
import Button from '../../src/common/components/Button';

jest.mock('../../src/auth/AuthService');

describe('LoginForm component', () => {
  beforeEach(() => {
    authService.login.mockClear();
  });

  it('should renders correctly', () => {
    const loginForm = renderer.create(<LoginForm />).toJSON();
    expect(loginForm).toMatchSnapshot();
  });

  it('should calls onLogin when user login', async () => {
    authService.login.mockResolvedValue();

    const { getByTestId, getByA11yLabel } = render(
      <LoginForm />,
    );

    const user = getByTestId('usernameInput');
    const pass = getByTestId('userPasswordInput');
    const button = getByA11yLabel('loginButton');

    await fireEvent.changeText(user, 'myuser');
    await fireEvent.changeText(pass, 'mypass');
    await fireEvent.press(button);

    // expect auth service login to be called once
    expect(authService.login).toBeCalledWith('myuser', 'mypass');
  });
});
