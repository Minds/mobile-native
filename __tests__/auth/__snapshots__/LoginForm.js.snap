// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component should renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#00000000",
        "elevation": 6,
        "overflow": "visible",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 20,
        "zIndex": 999,
      }
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "paddingTop": 10,
          },
          undefined,
          undefined,
          null,
          undefined,
          Object {
            "backgroundColor": "rgba(0,0,0,0.25)",
            "borderColor": "rgba(255,255,255,0.30)",
          },
        ]
      }
    >
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Roboto",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "marginBottom": 5,
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              Username
            </Text>
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCompleteType="username"
          containerStyle={
            Object {
              "backgroundColor": "rgba(0,0,0,0.25)",
              "borderColor": "rgba(255,255,255,0.30)",
            }
          }
          onChangeText={[Function]}
          placeholder=""
          placeholderTextColor="#444"
          rejectResponderTermination={true}
          returnKeyType="done"
          style={
            Array [
              undefined,
              Array [
                Object {
                  "borderWidth": 0,
                  "marginBottom": 7,
                  "paddingHorizontal": 0,
                  "paddingVertical": 8,
                },
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
          testID="usernameInput"
          textContentType="username"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View>
      <View
        style={
          Array [
            undefined,
            Object {
              "paddingTop": 10,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "rgba(0,0,0,0.25)",
              "borderColor": "rgba(255,255,255,0.30)",
            },
          ]
        }
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Roboto",
                      "fontSize": 16,
                      "fontWeight": "600",
                      "marginBottom": 5,
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                Password
              </Text>
            </View>
          </View>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCompleteType="password"
            containerStyle={
              Object {
                "backgroundColor": "rgba(0,0,0,0.25)",
                "borderColor": "rgba(255,255,255,0.30)",
              }
            }
            onChangeText={[Function]}
            placeholder=""
            placeholderTextColor="#444"
            rejectResponderTermination={true}
            returnKeyType="done"
            secureTextEntry={true}
            style={
              Array [
                undefined,
                Array [
                  Object {
                    "borderWidth": 0,
                    "marginBottom": 7,
                    "paddingHorizontal": 0,
                    "paddingVertical": 8,
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
            testID="userPasswordInput"
            textContentType="password"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 25,
            },
            Array [
              undefined,
              Object {
                "top": 26,
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ïŠ³
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityLabel="loginButton"
      disabled={false}
      onPress={[Function]}
      style={
        Array [
          undefined,
          undefined,
          Object {
            "backgroundColor": "rgba(0,0,0,0.40)",
            "borderColor": "rgba(255,255,255,0.60)",
            "borderRadius": 21,
            "paddingHorizontal": 23,
            "paddingVertical": 10,
          },
          Array [
            Object {
              "marginTop": 24.254545454545454,
            },
            undefined,
          ],
        ]
      }
      text="Login"
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 19,
            },
            Object {
              "color": "#FFFFFF",
            },
            undefined,
          ]
        }
      >
         
        Login
         
      </Text>
    </View>
    <View>
      <Text
        onPress={[Function]}
        style={
          Array [
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        Forgot your password?
      </Text>
    </View>
    <View />
    <View
      disabled={false}
      style={
        Array [
          undefined,
          undefined,
          Object {
            "backgroundColor": "rgba(0,0,0,0.40)",
            "borderColor": "rgba(255,255,255,0.60)",
            "borderRadius": 21,
            "paddingHorizontal": 23,
            "paddingVertical": 10,
          },
          Array [
            undefined,
            undefined,
          ],
        ]
      }
      testID="registerButton"
      text="Join Minds"
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 19,
            },
            Object {
              "color": "#FFFFFF",
            },
            undefined,
          ]
        }
      >
         
        Join Minds
         
      </Text>
    </View>
  </View>
</View>
`;
